### ISSUE TITLE
Supplying multiple properties to Unique decorator at entity level results in generation of redundant migrations

### ISSUE THREAD

**Issue by dystopiandev** [mikro-orm/mikro-orm#2932]
**Describe the bug**
Using `Unique` decorator with multiple properties (at entity level) causes `migration:create` to always generate migrations that just drop and add the same constraints, instead of reporting "schema is up-to-date".

**To Reproduce**
1. Apply `Unique` decorator to an entity. Ensure `properties` array has more than one field.
2. Run `migration:create` (with or without `--initial`).
3. Run `migration:up`.
4. Run `migration:create` again as many times over. It should keep generating redundant migrations on each run. Meanwhile if you supply just one property in the properties param of the `Unique` decorator on the entity, the expected behaviour is achieved.

**Expected behavior**
After generating and applying existing migration, `migration:create` should report `No changes required, schema is up-to-date` on subsequent runs.

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 14.19.0 |
| typescript | 4.6.2 |
| mikro-orm | 5.1.1 |
| mikro-orm/postgresql | 5.1.1 |

**Comment by B4nan** [comment]
Please provide reproduction as code (entity definition should be enough), there are tests for this and they are passing just fine. For example this entity has a compound unique index, used all over the ORM tests, including many schema specific tests that literally test no additional queries are produced:

https://github.com/mikro-orm/mikro-orm/blob/master/tests/entities-sql/Author2.ts#L13

**Comment by dystopiandev** [comment]
@B4nan I see the tests. Just figured the difference between those test cases and my issue is, one of the properties in my case is an Enum. Let me know if you still need repro, or this extra context is enough.

**Comment by B4nan** [comment]
Nope, still cannot reproduce it, tried both numeric and string enums, works fine on my end.

**Comment by dystopiandev** [comment]
Alright. I'll prepare a repro.

**Comment by B4nan** [comment]
Are you by any chance using some other naming strategy than the default one? If so, this would be a duplicate of #2938

**Comment by B4nan** [comment]
Let's reopen with a repro if the fix does not help.

**Comment by dystopiandev** [comment]
Hi. I've been away but I've just tried 5.1.2-dev.12 and still get redundant migrations.

My naming strategy extends `UnderscoreNamingStrategy` to pluralize table and key names, but I've also tested with the default naming strategy and still get the same results. Also, the concerned enum is numerical.

I'll prepare a repro.

**Comment by dystopiandev** [comment]
@B4nan  Here's the repro: [https://github.com/dystopiandev/mikro-orm-issue-2932-repro](https://github.com/dystopiandev/mikro-orm-issue-2932-repro)

**Comment by B4nan** [comment]
Minimal test case:

```ts
import { Entity, Enum, MikroORM, PrimaryKey, Unique } from '@mikro-orm/core';

export enum ChatLimitInterval {
  UNKNOWN = 0,
  INSTANT = 1,
  DAILY = 2,
  WEEKLY = 3,
  MONTHLY = 4,
}

@Entity()
@Unique({
  properties: ['interval', 'id'],
})
export class MessageThread {

  @PrimaryKey()
  id!: number;

  @Enum(() => ChatLimitInterval)
  interval!: ChatLimitInterval;

}

test('numeric enum diffing (GH issue #2932)', async () => {
  const orm = await MikroORM.init({
    entities: [MessageThread],
    dbName: `mikro_orm_test_enum1`,
    type: 'postgresql',
  });

  await orm.getSchemaGenerator().refreshDatabase();

  const diff = await orm.getSchemaGenerator().getUpdateSchemaSQL({ wrap: false });
  expect(diff).toBe('');

  await orm.close(true);
});
```

**Comment by dystopiandev** [comment]
> Minimal test case:
> 
> ```ts
> import { Entity, Enum, MikroORM, PrimaryKey, Unique } from '@mikro-orm/core';
> 
> export enum ChatLimitInterval {
>   UNKNOWN = 0,
>   INSTANT = 1,
>   DAILY = 2,
>   WEEKLY = 3,
>   MONTHLY = 4,
> }
> 
> @Entity()
> @Unique({
>   properties: ['interval', 'id'],
> })
> export class MessageThread {
> 
>   @PrimaryKey()
>   id!: number;
> 
>   @Enum(() => ChatLimitInterval)
>   interval!: ChatLimitInterval;
> 
> }
> 
> test('numeric enum diffing (GH issue #2932)', async () => {
>   const orm = await MikroORM.init({
>     entities: [MessageThread],
>     dbName: `mikro_orm_test_enum1`,
>     type: 'postgresql',
>   });
> 
>   await orm.getSchemaGenerator().refreshDatabase();
> 
>   const diff = await orm.getSchemaGenerator().getUpdateSchemaSQL({ wrap: false });
>   expect(diff).toBe('');
> 
>   await orm.close(true);
> });
> ```
> 
> The problem seems to be in order of columns in the constraint.

@B4nan  The above passes, but here's another (not sure if it is exactly the same issue): 
```ts
import {
  Entity,
  Enum,
  MikroORM,
  PrimaryKey,
  Property,
  Unique,
} from "@mikro-orm/core";

enum WalletInflowSource {
  UNKNOWN = 0,
  BTC_MAINNET = 1,
  ETH_MAINNET = 2,
}

enum WalletTransactionSettlementProvider {
  UNKNOWN = 0,
  RPC = 1,
}

@Entity()
@Unique({
  properties: ["inflowSource", "inflowReference"],
})
@Unique({
  properties: ["settlementProvider", "outflowSettlementReference"],
})
@Unique({
  properties: ["settlementProvider", "inflowSettlementReference"],
})
class WalletTransaction {
  @PrimaryKey()
  id!: number;

  @Enum(() => WalletTransactionSettlementProvider)
  settlementProvider!: WalletTransactionSettlementProvider;

  @Property()
  outflowSettlementReference?: string;

  @Property()
  inflowReference?: string;

  @Property()
  inflowSettlementReference?: string;

  @Enum(() => WalletInflowSource)
  inflowSource?: WalletInflowSource;
}

test("numeric enum diffing (GH issue #2932) [case 2]", async () => {
  const orm = await MikroORM.init({
    dbName: "mikro_orm_test_enum1",
    entities: [WalletTransaction],
    type: "postgresql",
  });

  await orm.getSchemaGenerator().refreshDatabase();

  const diff = await orm
    .getSchemaGenerator()
    .getUpdateSchemaSQL({ wrap: false });

  expect(diff).toBe("");

  await orm.close(true);
});
```

**Comment by B4nan** [comment]
Oh, truncated constraint names, ok let me try one more time :]
